### ISSUE TITLE
Self feedback data generation pipeline & reference model

### ISSUE THREAD

**Issue by charismaticchiu** [RLHF-V/RLAIF-V#6]
Hi 2 quick questions,

1. From the paper algorithm1, I get a sense that the algorithm can work in an online divide-n-conquer manner with updated model and I am just curious when the self-feedback code will be released.

2. In [this line](https://github.com/RLHF-V/RLAIF-V/blob/main/muffin/train/train_llava15.py#L165) the reference model is initialized the same as policy model (weights and dtype), so why wouldn't the log_p be the same and thus the loss be 0 all the time?

Thank you!

**Comment by Haoye17** [comment]
Hi @charismaticchiu,

Thank you for your interest in our work! To address your questions:

1. **Self-feedback Code**: We have updated our code for generating LLaVA 1.5 feedback data using OmniLMM and MiniCPM-Llama3-V 2.5. You can find detailed instructions for data generation [here](https://github.com/RLHF-V/RLAIF-V?tab=readme-ov-file#data-generation).
2. **DPO Optimization**: Regarding the DPO optimization, because the loss calculation involves a log sigmoid function externally, the loss will not be zero even if the policy and reference models are identical.
    <img width="866" alt="image" src="https://github.com/RLHF-V/RLAIF-V/assets/42668396/ebab3d60-496d-4b78-956b-df3893949559">

I hope this helps! If you have any other questions, feel free to ask.

**Comment by charismaticchiu** [comment]
Thanks for the quick reply! Make sense!


Btw, where can I find the question file in [this line](https://github.com/RLHF-V/RLAIF-V/blob/main/script/data_gen/run_data_pipline_llava15_omni.sh#L6)?

**Comment by Haoye17** [comment]
Hi @charismaticchiu,

You can find the question file [here](https://github.com/RLHF-V/RLAIF-V/blob/main/examples/detail_test_30_input.jsonl)~

**Comment by charismaticchiu** [comment]
Awesome, thanks!

and in the paper the model is trained 4 epochs, but the [script](https://github.com/RLHF-V/RLAIF-V/blob/f6a6b82db70a94d97f904c58d4c8e2a373916309/script/train/llava15_train.sh#L20) shows it was trained 10 epochs, was that a typo?  Also I am a bit confused by the [steps](https://github.com/RLHF-V/RLAIF-V/blob/main/script/train/llava15_train.sh#L30), which overrides the `num_train_epochs`. Does that match the implementation details right before sec 3.2?

**Comment by yiranyyu** [comment]
1. We use `max_steps` to control how many training/optimization steps are needed, rather than `num_train_epochs`. Actually, we also conducted some further experiments with this codebase before releasing. So the default hyper-parameters written in scripts might not all match what shown in the paper. For reproduction, we refer users to the detail introduction in the paper.
2. The scripts utilizing LLaVA-NeXT as the labeler model are still being prepared for release. Please stay tuned for updates. In fact, you can simply change the inference code in provided code with those of LLaVA-NeXT for fast adaptation.

**Comment by yiranyyu** [comment]
No, it does not have any special meaning, it is just because the number of steps per epoch is not a pretty number like 1024 or 1500. Feel free to open a new issue if you have any questions.

**Comment by lyklly** [comment]
To supplement the previous answer, the number of data in 006.parquet is 5342, so the calculation method of 2672 is len(dataset) / ( gradient_accumulation_steps * batch_size) rounded up and multiplied by num_epoch (that is, 4), which is equal to 668*4=2672. That should be right, right?
